src/scripts/mentoring-data-migration.js

```diff
@@ -0,0 +1,1127 @@
+		console.log(`\n🔄 Processing user_extensions with organization_id lookup...`)
+
+		// For user_extensions, we need to get organization_id from user service data
+		// This requires a different approach - we'll use the individual record processing
+		await this.processTable(tableConfig)
+	}
+
+	/**
+	 * Process table with user_id using user_extensions updated data
+	 */
+	async processTableWithUserId(tableConfig) {
+		const { name, updateColumns, hasPartitionKey } = tableConfig
+		console.log(`\n🔄 Processing table with user_id: ${name}`)
+
+		try {
+			// Check if table exists and has target columns
+			const existingColumns = await this.checkTableColumns(name)
+			const availableUpdateColumns = updateColumns.filter((col) => existingColumns.includes(col))
+
+			if (availableUpdateColumns.length === 0) {
+				console.log(`⚠️  Table ${name} has no target columns, skipping`)
+				return
+			}
+
+			// Only update tenant_code for user_id tables
+			if (availableUpdateColumns.includes('tenant_code')) {
+				const citusEnabled = await this.isCitusEnabled()
+
+				// Undistribute if needed
+				let wasDistributed = false
+				if (citusEnabled && hasPartitionKey) {
+					wasDistributed = await this.undistributeTable(name)
+				}
+
+				try {
+					await this.executeUserIdUpdates(name)
+
+					// Add detailed logging like in migration
+					await this.logTableUpdateResults(name)
+
+					// Redistribute if needed
+					if (citusEnabled && wasDistributed) {
+						await this.redistributeTable(name)
+					}
+				} catch (error) {
+					console.error(`❌ Error during updates for ${name}:`, error.message)
+					throw error
+				}
+			}
+
+			console.log(`✅ Completed ${name}`)
+		} catch (error) {
+			console.error(`❌ Error processing table ${name}:`, error)
+			throw error
+		}
+	}
+
+	/**
+	 * Execute updates for tables with user_id using user_extensions data
+	 */
+	async executeUserIdUpdates(tableName) {
+		console.log(`📊 Executing user_id updates for ${tableName}...`)
+
+		const transaction = await this.sequelize.transaction()
+
+		try {
+			// Get table configuration
+			const tableConfig = this.tablesWithUserId.find((t) => t.name === tableName)
+
+			let updateQuery
+
+			if (tableConfig.useSessionLookup) {
+				// Special case for post_session_details: get data from sessions table based on session_id
+				const sessionIdColumn = tableConfig.columns.session_id
+
+				// Build SET clause based on available update columns
+				const setClauses = []
+				if (tableConfig.updateColumns.includes('tenant_code')) {
+					setClauses.push('tenant_code = s.tenant_code')
+				}
+				if (tableConfig.updateColumns.includes('organization_code')) {
+					setClauses.push('organization_code = s.organization_code')
+				}
+				setClauses.push('updated_at = NOW()')
+
+				// Build WHERE conditions based on what columns sessions table actually has
+				const whereConditions = [`${tableName}.${sessionIdColumn} = s.id`, 's.tenant_code IS NOT NULL']
```

Comment:
@sumanvpacewisdom as you are setting up by the tenantCode as default value, this won't work it seems